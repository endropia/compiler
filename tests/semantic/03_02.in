type arr = array[0..10] of integer;
type arr_of_arr = array[0..10] of array [0..10] of integer;
type arr_ = array[0..10, 0..10, 0.. 10] of integer;
begin end.