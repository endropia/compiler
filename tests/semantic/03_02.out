program : Unnamed program
   alias
      array
      type: integer
      range
         0
         10
      arr
   alias
      array
      array
         type: integer
         range
            0
            10
      range
         0
         10
      arr_of_arr
   alias
      array
      type: integer
      range
         0
         10
      range
         0
         10
      range
         0
         10
      arr_
   stmts:
      empty
scope     name                          class               
------------------------------------------------------------
0         integer                       primitive type      
0         double                        primitive type      
0         boolean                       primitive type      
0         char                          primitive type      
0         string                        primitive type      
1         arr                           alias               
1         arr_of_arr                    alias               
1         arr_                          alias               
